src/lib/message/L2ToL1MessageNitro.ts

```diff
@@ -155,6 +157,10 @@ export class L2ToL1MessageReaderNitro extends L2ToL1MessageNitro {
     return outbox.callStatic.isSpent(this.event.position)
   }
 
+  public setL2BlockRangeCache(l1Block: number, l2BlockRange: number[]) {
+    l2BlockRangeCache[l1Block] = l2BlockRange
+  }
```

Comment:
I would remove this method and just directly update the cache above from the code